### PR TITLE
subscription: MailChimp replaced for Mailtrain

### PR DESCRIPTION
**Pull request by pazembrz**
INSPIR-2676

### CODE REVIEW

File: backend/inspirehep/mailing/providers/mailtrain.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+def mailtrain_subscribe_user_to_list(list_id, email, first_name, last_name):
+    """Subscribe user to a list.
+
+    Args:
+        list_id (str): the list id .
+        email (str): user email.
+        first_name (str): user first name.
+        last_name (str): user last name.
+    """
+    LOGGER.info("Adding user to the mailing list.", list_id=list_id, email=email)
```
Comment (ammirate) [inline backend/inspirehep/mailing/providers/mailtrain.py]: Not sure we should explicitly log email addresses.

File: backend/inspirehep/mailing/providers/mailtrain.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    if response.status_code != 200:
+        LOGGER.error(
+            "Cannot add user to MailTrain! Error: %s. Status_code: %s",
+            response.reason,
+            response.status_code,
+        )
+        raise ValueError(response.reason)
+    else:
+        LOGGER.info(
+            "User successfully added to the mailing list.", list_id=list_id, email=email
```
Comment (ammirate) [inline backend/inspirehep/mailing/providers/mailtrain.py]: same here

File: backend/inspirehep/mailing/providers/mailtrain.py (hunk 1 above)
Comment (pazembrz) [reply backend/inspirehep/mailing/providers/mailtrain.py]: I am also not sure. It was done like that before for previous user submission.

File: backend/inspirehep/mailing/providers/mailtrain.py (hunk 2 above)
Comment (pazembrz) [reply backend/inspirehep/mailing/providers/mailtrain.py]: same here

**Pull request merged**